File: frontend/src/objects/order/core/OrderSummaryPricing.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type { CurrencyCents } from '@/objects/core/SharedObjects'
import type { Coupon } from '@/objects/customer/profile/Coupon'

export type OrderSummaryPricing = {
  itemSubtotalCents: CurrencyCents
  deliveryFeeCents: CurrencyCents
  couponDiscountCents: CurrencyCents
  appliedCoupon?: Coupon
  totalPriceCents: CurrencyCents
}

File: backend/src/main/scala/domain/order/OrderSummaryPricing.scala
package domain.order

import domain.customer.Coupon
import domain.shared.*

final case class OrderSummaryPricing(
    itemSubtotalCents: CurrencyCents,
    deliveryFeeCents: CurrencyCents,
    couponDiscountCents: CurrencyCents,
    appliedCoupon: Option[Coupon],
    totalPriceCents: CurrencyCents,
)
